### PR TITLE
refactor(notebook): share notebook view projections

### PR DESCRIPTION
**Pull request by quillaid**
## Summary
- Extract a host-neutral NotebookView store projector for materialized cells, synthetic outputs/executions, markdown projection, cache-key reuse, and projector-owned cleanup.
- Replace duplicated desktop/cloud rail and outline selection React state with a shared useSyncExternalStore-backed rail UI store.
- Update cloud/desktop wiring tests and the shared-store projection ADR to record what landed and what remains.

## Verification
- pnpm test:run src/components/notebook/__tests__/view-store-projection.test.ts src/components/notebook/__tests__/rail-ui-state.test.ts src/components/notebook/__tests__/outline-interaction.test.ts apps/notebook/src/components/__tests__/notebook-toolbar.test.tsx
- pnpm --dir apps/notebook-cloud exec node --import tsx --test test/cloud-projection-flicker.test.ts test/notebook-view-store-bridge.test.ts test/viewer-shared-cell-surface.test.ts test/viewer-render-props.test.ts
- pnpm --dir apps/notebook exec tsc -b --pretty false
- pnpm --dir apps/notebook-cloud typecheck
- cargo xtask lint --fix

### CODE REVIEW

**Comment by rgbkrk** [comment]
@pullfrog review

File: apps/notebook-cloud/viewer/notebook-view-store-bridge.ts [hunk 1]
```diff
@@ -22,127 +7,22 @@ import {
...
-let cloudOwnedExecutionIds = new Set<string>();
-let cloudOwnedOutputIds = new Set<string>();
-
-const RUNT_OUTPUT_CACHE_KEY = "_runt_output_cache_key";
-const outputCacheKeys = new WeakMap<NotebookStoreOutput, string>();
+const cloudViewStoreProjector = createNotebookViewStoreProjector({
+  syntheticExecutionId: (cellId) => `cloud-execution:${cellId}`,
+  syntheticOutputId: (cellId, outputIndex) => `cloud-output:${cellId}:${outputIndex}`,
+  syntheticOutputPrefix: (cellId) => `cloud-output:${cellId}:`,
+});
```
Comment (rgbkrk) [inline apps/notebook-cloud/viewer/notebook-view-store-bridge.ts]: Why do these have cloud prefixes? Can't these functions be the same in both desktop and cloud?

File: apps/notebook-cloud/viewer/notebook-view-store-bridge.ts [hunk 2]
```diff
@@ -22,127 +7,22 @@ import {
...
-  setNotebookQueueProjection({
-    executing_cell_id: null,
-    queued_cell_ids: [],
-  });
-  deleteOutputs([...cloudOwnedOutputIds]);
-  deleteExecutions([...cloudOwnedExecutionIds]);
-  cloudOwnedOutputIds = new Set<string>();
-  cloudOwnedExecutionIds = new Set<string>();
+  cloudViewStoreProjector.reset({ resetQueue: true });
 }
```
Comment (rgbkrk) [inline apps/notebook-cloud/viewer/notebook-view-store-bridge.ts:22]: Great seeing this all unify

**Comment by quillaid** [comment]
Addressed both review points in 4272ee4a.

- Removed the cloud-specific synthetic ID factories from `notebook-view-store-bridge.ts`; cloud now uses the shared `createNotebookViewStoreProjector()` defaults (`notebook-view-*`) just like any other host adapter.
- Added RuntimeStateDoc ownership tracking in the execution store. `applyExecutionViewChangeset` marks runtime-authored execution IDs, and the projector now releases placeholder ownership without deleting a runtime-owned record if RuntimeStateDoc later authors the same ID.
- Updated the ADR and tests for the shared default IDs and the runtime-ownership edge case.

Verification rerun:
- `pnpm test:run src/components/notebook/__tests__/view-store-projection.test.ts apps/notebook/src/lib/__tests__/project-runtime-stores.test.ts src/components/notebook/__tests__/rail-ui-state.test.ts src/components/notebook/__tests__/outline-interaction.test.ts apps/notebook/src/components/__tests__/notebook-toolbar.test.tsx`
- `pnpm --dir apps/notebook-cloud exec node --import tsx --test test/cloud-projection-flicker.test.ts test/notebook-view-store-bridge.test.ts test/viewer-shared-cell-surface.test.ts test/viewer-render-props.test.ts`
- `pnpm --dir apps/notebook exec tsc -b --pretty false`
- `pnpm --dir apps/notebook-cloud typecheck`
- `cargo xtask lint --fix`

File: apps/notebook-cloud/viewer/notebook-view-store-bridge.ts (hunk 1 above)
Comment (quillaid) [reply apps/notebook-cloud/viewer/notebook-view-store-bridge.ts]: Yes, agreed. I removed the cloud-specific projector factories in 4272ee4a so the bridge now calls `createNotebookViewStoreProjector()` directly and uses the shared `notebook-view-*` synthetic placeholders. The remaining cloud bridge code is just bootstrap-preservation policy.
